### DISCUSSION TITLE
footnotes sorting script

### DISCUSSION THREAD

**Discussion by hellt** [squidfunk/mkdocs-material#7496] [Q&A] [answered]
Hi everyone,
has anyone come up with a script that would sort the footnotes in a document?

E.g. when you have smth like this:

```markdown
some text[^1]

more text[^2]

[^1]: note 1
[^2]: note 2
```

and then you add another line inbetween

```markdown
some text[^1]

new line[^3]

more text[^2]

[^1]: note 1
[^2]: note 2
[^3]: note 3
```

your footnotes become unsorted. OCD kicks in...

**Comment by squidfunk** [comment]
You know that you can use any identifier to link to footnotes? I'm also mostly using numbers, but you can use `^whateveryoulike`, which might appeal more to your OCD 😅 I'm on the same boat, btw.

**Reply by hellt** [reply]
yeah, what trips me is the ordering in the rendered superscript (which is always a number, not configurable in mkdocs-material), because it will be unordered as you read the document.

I think what I'd likely appreciate more is the annotation icon as the superscript. E.g. the same (+) sign, but as a superscript so that it doesn't stand out as much as its inline version.

**Comment by hellt** [comment, marked as answer]
Okay, last night I decided to google the footnote sorter examples, expecting that there should've been poor souls suffering from this.

I found a 11 y.o project, which I refactored to support rich footnote content and different styling of footnotes.

https://github.com/hellt/markdown-footnote-sorter

Usage:

```bash
docker run --rm -v $(pwd):/work ghcr.io/hellt/markdown-footnote-sorter path/to/doc.md
```


https://github.com/user-attachments/assets/6b817dd9-c7da-4d40-af67-58c4dbcae0cc

**Reply by squidfunk** [reply]
Nice, great work!
